toqito/measurement_ops/measure.py

```diff
@@ -1,76 +1,118 @@
 
-    Further information can be found here :cite:`WikiQuantMeas`.
+    when `update` is True. Then, for each operator :math:`K_i`, the outcome probability is computed as
 
-    Examples
-    ==========
+    .. math::
+       p_i = \mathrm{Tr}\Bigl(K_i^\dagger K_i\, \rho\Bigr),
 
-    Consider the following state:
+    and, if :math:`p_i > tol`, the post‐measurement state is updated via
 
     .. math::
-        u = \frac{1}{\sqrt{3}} e_0 + \sqrt{\frac{2}{3}} e_1
-
-    where we define :math:`u u^* = \rho \in \text{D}(\mathcal{X})`.
+       \rho_i = \frac{K_i\, \rho\, K_i^\dagger}{p_i}.
 
-    Define measurement operators
+    If :math:`p_i \le tol`, the corresponding post‐measurement state is a zero matrix.
 
-    .. math::
-        P_0 = e_0 e_0^* \quad \text{and} \quad P_1 = e_1 e_1^*.
+    Examples
+    ========
+    **Single operator (POVM element):**
 
-    >>> from toqito.states import basis
-    >>> from toqito.measurement_ops import measure
-    >>> import numpy as np
-    >>> e_0, e_1 = basis(2, 0), basis(2, 1)
-    >>>
-    >>> u = 1/np.sqrt(3) * e_0 + np.sqrt(2/3) * e_1
-    >>> rho = u @ u.conj().T
-    >>>
-    >>> proj_0 = e_0 @ e_0.conj().T
-    >>> proj_1 = e_1 @ e_1.conj().T
+    .. jupyter-execute::
 
-    Then the probability of obtaining outcome :math:`0` is given by
+        import numpy as np
+        from toqito.measurement_ops.measure import measure
+        rho = np.array([[0.5, 0.5], [0.5, 0.5]])
+        proj_0 = np.array([[1, 0], [0, 0]])
 
-    .. math::
-        \langle P_0, \rho \rangle = \frac{1}{3}.
+        # Without update; simply returns the probability.
+        print(measure(rho, proj_0))
 
-    >>> measure(proj_0, rho)
-    0.3333333333333334
+        # With state update; returns (probability, post_state).
+        p, post_state = measure(rho, proj_0, update=True)
+        print(p)
 
-    Similarly, the probability of obtaining outcome :math:`1` is given by
+    **Multiple operators (Kraus operators):**
 
-    .. math::
-        \langle P_1, \rho \rangle = \frac{2}{3}.
+    .. jupyter-execute::
 
-    >>> measure(proj_1, rho)
-    0.6666666666666666
+        import numpy as np
+        from toqito.measurement_ops.measure import measure
+        rho = np.array([[0.5, 0.5], [0.5, 0.5]])
+        K0 = np.array([[1, 0], [0, 0]])
+        K1 = np.array([[0, 0], [0, 1]])
 
-    References
-    ==========
-    .. bibliography::
-        :filter: docname in docnames
+        # Returns list of probabilities.
+        print(measure(rho, [K0, K1]))
 
+        # Returns list of (probability, post_state) tuples.
+        print(measure(rho, [K0, K1], update=True))
 
-    :param measurement: The measurement to apply.
-    :param state: The state to apply the measurement to.
-    :return: Returns the probability of obtaining a given outcome after applying
-             the variable :code:`measurement` to the variable :code:`state`.
+    :param state: Quantum state as a density matrix (an np.ndarray with shape (d, d)).
```

Comment:
```suggestion
    :param state: Quantum state as a density matrix shape (d, d) where d is......
```
No need to add type here but you should describe what d is.